### ISSUE TITLE
No Polling Time for older Stove Model

### ISSUE THREAD

**Issue by Voidfloop** [Crisicus/maestro_mcz#114]
No polling time for older stoves. Polling only works with manual entries such as: timer switch, fan control etc.
If you hear a beep, there is new data.

![IMG_7848](https://github.com/Robbe-B/maestro_mcz/assets/146527945/e12394a4-eb53-4ab3-bbad-60b247c5688d)

**Comment by Robbe-B** [comment]
@Voidfloop,
 is there any error in the logging ?

**Comment by Voidfloop** [comment]
Do you mean in the Home Assistant Core log file?
There is no error recorded there.
There were problems with the installation yesterday. Uninstalling and reinstalling worked.
But during normal operation there is nothing in the log file.

**Comment by Voidfloop** [comment]
The only quiet push I have is to turn the beep off.
I could trigger this via automation.
But then there are no more acknowledgment tones.
Maybe there is another way to trigger the stove?

**Comment by Robbe-B** [comment]
> The only quiet push I have is to turn the beep off. I could trigger this via automation. But then there are no more acknowledgment tones. Maybe there is another way to trigger the stove?

Maybe, the thing is, I can't really know that since I don't have an older stove 😞

**Comment by Voidfloop** [comment]
So I'm currently testing my workaround.
I trigger the beep silent mode every minute.
I have concerns about the lifespan of the installed eprom. I have no idea whether it has problems with being constantly written to, or whether that even happens with the command.
I don't delve deep enough into the matter.

I will help you as best I can to improve your integration

`alias: Pelletofen Trigger
description: ""
trigger:
  - platform: time_pattern
    minutes: "1"
condition: []
action:
  - service: select.select_option
    metadata: {}
    data:
      option: Silent
    target:
      entity_id:
        - select.wohnzimmer_tones
        - select.keller_tones
mode: single
`

**Comment by Robbe-B** [comment]
@Crisicus, are you aware of M1 models using another "ping" call ?

**Comment by Crisicus** [comment]
@Robbe-B I do not.
There's probably a method for the M1 stoves, but without having access to one of those stoves, I don't think it'll be easy to find.

@Voidfloop I know this is a stretch, and I obviously understand if you don't want to, but if you could share your stove with me in the app, I could check what the app uses to poll your stove type.

**Comment by Robbe-B** [comment]
If there is any user reading this, 
who happens to have a M1+ model stove and is willing to let us have a look at your stove. (Credentials can be changed afterwards).
This is obviously a lot to ask, but without this, we cannot move forward on this topic.

NOTE: For the more technical folks who have such model. 
The technical information would be a great help as well.
Kind regards, 
Robbe

**Comment by Robbe-B** [comment]
@Crisicus, 
I noticed an app update related to performance,
Any chance you can find out what they've actually changed to achieve this ? I wonder if we should do the same for this integration then.
![Screenshot_2024-09-12-14-53-53-09_b5a5c5cb02ca09c784c5d88160e2ec24.jpg](https://github.com/user-attachments/assets/b79fa09d-d3b8-418a-882d-b96e7d76890e)


Kind regards,
Robbe

**Comment by Voidfloop** [comment]
First of all, sorry that I haven't been in touch for a long time.
I have to do some maintenance on my stove first.
The best thing to do is to log on to my stove and see for yourself what's happening with the communication.
Can't you write a direct message here because of the login details?
Perhaps a cell phone number via WhatsApp.
Direct communication would be better. I don't want to have the stove on all the time, the weather is still relatively warm. ;)

**Comment by Voidfloop** [comment]
I just turned the stove on and noticed that automatic polling is still not being carried out. Only when something is manually set again do the current values ​​appear.

**Comment by Robbe-B** [comment]
@Voidfloop, 
Great to hear from you again ! 
It would indeed be a great help if we would be able to login to your stove in order to figure out how this exactly works.
This is something @Crisicus is more comfortable with as far as investigating goes. 
So @Crisicus, if you are here, would you mind getting a look at this ? 
@Voidfloop, maybe try to get in touch with @Crisicus, he will provide me with the details in order to implement this.
It's nice collaborating with you guys ! 
Kind regards,
Robbe

**Comment by pvillmann** [comment]
Hi, if you Need a second stove to Check something just contact me.
My stove is a Club Hydromatic M1 from 2020.

**Comment by Robbe-B** [comment]
@Crisicus ^^^^

**Comment by Crisicus** [comment]
I'l currently in the USA for the rest of the month. Will check back afterwards!

**Comment by Robbe-B** [comment]
@Crisicus, 
How are you doing ? 
Did you already had some time to have a look at this ? 
Actually more and more people are asking me about this.

Kind regards,
Robbe

**Comment by Robbe-B** [comment]
@htbuddha,
In https://github.com/Robbe-B/maestro_mcz/issues/162, you where talking about the "upgrade" app.
How do you go from the "dumb" app to the "upgrade" app ? 
Because you need an MCZ account at some point right ?

**Comment by htbuddha** [comment]
You simply download the upgrade app register an MCZ account (which you previously did not have, with the "dumb" app)

**Comment by Robbe-B** [comment]
I might try to get a used motherboard of such model. 
I'll see if this is possible.

**Comment by Robbe-B** [comment]
The only workaround I can think of is setting the system date and time every 30 seconds via an automation.
This will probably let the polling kick back in ? 
![image](https://github.com/user-attachments/assets/c1f738a6-ab73-42ce-b8b9-7b43882d00cc)

**Comment by htbuddha** [comment]
> The only workaround I can think of is setting the system date and time every 30 seconds via an automation. This will probably let the polling kick back in ? ![image](https://private-user-images.githubusercontent.com/103053873/387709424-c1f738a6-ab73-42ce-b8b9-7b43882d00cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMjg4MTQsIm5iZiI6MTczMjAyODUxNCwicGF0aCI6Ii8xMDMwNTM4NzMvMzg3NzA5NDI0LWMxZjczOGE2LWFiNzMtNDJjZS1iOGI5LTdiNDM4ODJkMDBjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOVQxNTAxNTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YmIxNDBmNTg5YzI3MmYxZDk5YzY3MGI1ZDFhNTY3MTBmNWUzY2U4YzQ4ODhiYTdlZWY2N2I0YmFlMzU2NWJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bgiLs5X3nNG1qFYSzyFbut7Mq5goOlf_dBKwlZ-22Q8)

Yes, either that or setting the sound option to "silent" every 30 seconds. An automation on iPhone (shortcuts) where you open the application every 30 seconds should work too, but obviously  causes battery drain.

**Comment by Robbe-B** [comment]
Anyway, I'll search for a more permanent solution in the long term.
Once this issue is solved, this integration might be the best one for HA, sadly it's still using there wonky cloud infrastructure.

**Comment by htbuddha** [comment]
Yes. It would be much better if you could use the local connection to read the data. Idk if that works on the newer models, but the older stoves can be controlled locally. Perhaps add a drop down menu for users to select old models or new models, and for old models make use of the local connection.

**Comment by Robbe-B** [comment]
> Yes. It would be much better if you could use the local connection to read the data. Idk if that works on the newer models, but the older stoves can be controlled locally. Perhaps add a drop down menu for users to select old models or new models, and for old models make use of the local connection.

It's a good suggestion but it's not that simple like just adding a dropdown, the local communication is using an entire other protocol to communicate from my understanding. Also the local communication need to be done over a completely separate connection => "WIFI HOME" SSID the stove sends out.

In the newer models they completely replaced this with Bluetooth Low Energy (BLE) which makes it even harder to understand the workings and to locally communicate with these stove models without extra hardware like the HA Bluetooth Proxies.

**Comment by htbuddha** [comment]
![image](https://github.com/user-attachments/assets/a3461d29-4fd2-433f-abcc-309c1c1e2f57)
Unfortunately the date and time workaround does not seem to be solving the issue. The ambient temperature is not being updated correctly.

**Comment by Robbe-B** [comment]
@htbuddha, 
FYI => I got my hands on a MCZ TILDA Comfort Air M1 10 kW
I've added this one into HA with this integration, but it seems to update fine (without any apps open).
![image](https://github.com/user-attachments/assets/5163aa69-a018-48e8-a099-56b325f4a20d)
This might be the same as it worked fine for you as well in the beginning.

**Comment by htbuddha** [comment]
Thanks for the update. Very odd. What's even more odd is that this morning I woke up to a dead stove. The technician just came and the motherboard/"printplaat" is broken according to him. And they don't have any in stock...


I wonder what causes a motherboard to die like that.

**Comment by Robbe-B** [comment]
I found something interesting in the config of the M1 models (MCZ TILDA Comfort Air M1 10 kW)=> 
![image](https://github.com/user-attachments/assets/41845d59-dc2c-40d3-be6c-2b4baf009b71)

**Comment by htbuddha** [comment]
They have finally installed a new motherboard. The same issue persists. 
This config that you show, where was it gathered from? Was this included in a recent update?
Would make sense, and conform my analysis of the issue.

**Comment by Robbe-B** [comment]
@htbuddha,
I've got this from the diagnostics file from this integration for the "MCZ TILDA Comfort Air M1 10 kW" model.
But it implies the ping command is maybe implemented different for the M1 models.
Instead of the "ping" method provided by the API.
@Crisicus any thoughts on this ?
The M1 models (MCZ upgrade app) probably sets this as a 'configuration' (like all other sets in the integration) rather then using the dedicated ping method.
Kind regards, 
Robbe

**Comment by Crisicus** [comment]
@Robbe-B 
These configs are for the older models with a WiFi AP. They are the commands you can send over websocket to the stove.

I saw these first when researching on this repo; https://github.com/Chibald/maestrogateway

**Comment by Robbe-B** [comment]
@Crisicus, 
Indeed, these models can now also use this integration once you connect them from the "MCZ Maestro - Upgrade" app (the gray one)
[<img src="https://github.com/Robbe-B/maestro_mcz/blob/main/docs/old_app_icon.png" width="128" height="128">](https://github.com/Robbe-B/maestro_mcz/raw/main/docs/old_app_icon.png)

You can therefore migrate from the old MCZ app to that upgrade app which let you create an account.
From that point onwards you can simply start using this integration.

**Comment by Robbe-B** [comment]
@Crisicus, 
The only thing missing is the correct ping command.
Would you be able to find this out in the "MCZ Maestro - Upgrade" app ?

**Comment by htbuddha** [comment]
> I've got this from the diagnostics file from this integration for the "MCZ TILDA Comfort Air M1 10 kW" model.
> But it implies the ping command is maybe implemented different for the M1 models.
> Instead of the "ping" method provided by the API.

How would that explain that it used to work perfectly for months, and then suddenly stopped working?
Looking forward to the progress and hopefully you can fix the issue. That'd be great!

**Comment by htbuddha** [comment]
> > > I've got this from the diagnostics file from this integration for the "MCZ TILDA Comfort Air M1 10 kW" model.
> > 
> > 
> > > But it implies the ping command is maybe implemented different for the M1 models.
> > 
> > 
> > > Instead of the "ping" method provided by the API.
> > 
> > 
> > How would that explain that it used to work perfectly for months, and then suddenly stopped working?
> > Looking forward to the progress and hopefully you can fix the issue. That'd be great!
> 
> From what we've seen so far, MCZ changes their API and way of working a lot. Keep in mind that everything in this and other repo's is 100% unofficial, MCZ does not offer public API's, and thus, have no obligation to prevent breaking changes. We will always be 1 step behind until they release a public API, if they ever choose to do so.

I'm aware and did not point any fingers or blame you(?)
Just looked for confirmation whether or not that change was indeed introduced with an update.

**Comment by htbuddha** [comment]
Hi, just checking whether or not there is any news regarding this bug?
Thanks in advance.

**Comment by Elildar** [comment]
Hello, I have a M1 stove and also have this bug.
My workaround is to set the target temperature every minute. I then receive the current status of the stove.
If I don't do that, I rarely receive its status and ambient temperature.

I started with the following project : https://github.com/giuseppeg88/node-red-contrib-thermostat-scheduler
And improved it to add compatibility with MCZ stoves and to meet my expectations. Now, it replaces the Chronostato parameters and reduce the number of time the stove starts.

**Comment by Robbe-B** [comment]
No update so far,
I'm aware of the bug. 
But without knowing the exact ping command older models are using, there is nothing I can do ....
Kind Regards, 
Robbe
